connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -521,6 +525,17 @@ ble_error_t Gap::connect(
 
     ble_error_t ret = BLE_ERROR_INTERNAL_STACK_FAILURE;
 
+#if BLE_GAP_HOST_BASED_PRIVATE_ADDRESS_RESOLUTION
+    if (_ready_to_connect_to_host_resolved_address) {
```

Comment:
It took me some time to understand what this exactly does. Using a state enum would make the code more readable and prevent multiple calls to `connect` that may lead to memory leak.